File: 001-two_sum/two_sum.py
#!/usr/bin/env python3

class Solution:
    def twoSum(self, nums, target):
        """
        type nums: List[int]
        type target: int
        r type: List[int] 
        """
        dic = {}
        for index, num in enumerate(nums):
            another_num = target - num
            if another_num in dic:
                return [dic[another_num], index]
            dic[num] = index
        return None
        
nums = [1, 3, 5, 7]
target = 6

result = Solution()
print(result.twoSum(nums, 6))

File: 001-two_sum/two_sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
    lenNums := len(nums)
    
    for i:=0; i < lenNums; i++ {
        for j:=i+1; j < lenNums; j++ {
            if nums[i] + nums[j] == target {
                return []int{i, j}
            }
        }
    }
    
    return []int{}
}

func main() {
    var nums  = []int{1, 2, 3, 5}
    var result []int
    var i int
    target := 5
    
    result = twoSum(nums, target)
    for i = 0; i < 2; i++ {
        fmt.Println(result[i])
    }
}